### PR TITLE
Issue#13345: Enable ilegalTokenTextExamplesTest

### PR DESCRIPTION
**Pull request by Coding-Aliens**
Issue: #13345 
Sorry for deleting the branch causing the last PR to close.
1. In example2, the program failed to detect violation message with explanation, problem may be in src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
2. In example3, text block will cause compile error in source10 environment, I turned it to another pattern and thus, test disabled

### CODE REVIEW

**Comment by romani** [comment]
Please investigate CI failures and fix them

**Comment by romani** [comment]
There should be changes in xdoc xml files, please recheck why it is not happening.

>problem may be in src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

Most likely in Input files.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java [hunk 1]
```diff
@@ -9,9 +9,11 @@
 </module>
 */
 
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltokentext;
 // xdoc section -- start
-public void myTest() {
-    final String quote = """
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java:14]: Please restore it , and put this file to non compiled resources folder, we have bunch of examples already

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java (hunk 1 above)
Comment (Coding-Aliens) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java]: Put it to the non-complied folder as requested, xml file edited

**Comment by Coding-Aliens** [comment]
Last four similar errors remaining, example: 
PropertyCacheFileTest.testMultipleSymbolicLinkResolution:350 » FileSystem C:\Users\kendr\AppData\Local\Temp\firstLinkDir4001639906560686789\firstLink: A required privilege is not held by the client
Now working on it

**Comment by romani** [comment]
CI failures:
```
[INFO] [checkstyle] Running Checkstyle  on 2111 files
[ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\src\xdocs-examples\java\com\puppycrawl\tools\checkstyle\checks\coding\IllegalTokenTextCheckExamplesTest.java:23:1: 'org.junit.jupiter.api.Test' should be separated from previous import group by one line. [CustomImportOrder]
[ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\src\xdocs-examples\java\com\puppycrawl\tools\checkstyle\checks\coding\IllegalTokenTextCheckExamplesTest.java:23:1: 'org.junit.jupiter.api.Test' should be separated from previous imports. [ImportOrder]
```

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java [hunk 2]
```diff
@@ -8,10 +8,15 @@
...
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltokentext;
+
+public class Example3 {
+    public void myTest() {
+        final String quote = """
+                   \"""";
+    }
 }
-// xdoc section -- end
+// xdoc section -- end
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java:22]: add line ending.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java [hunk 3]
```diff
@@ -9,9 +9,13 @@
...
-    String test = "a href"; // violation
-    String test2 = "A href"; // OK, case is sensitive
+public class Example1 {
+    public void myTest() {
+        String test = "a href"; // violation
+        String test2 = "A href"; // OK, case is sensitive
+    }
 }
-// xdoc section -- end
+// xdoc section -- end
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java:21]: add line ending

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java [hunk 4]
```diff
@@ -8,10 +8,15 @@
...
+//non-compiled with javac: Compilable with Java14
 
 // xdoc section -- start
-public void myTest() {
-    final String quote = """
-               \""""; // violation
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltokentext;
+
+public class Example3 {
+    public void myTest() {
```
Comment (romani) [inline src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java:17]: please make indentation as +2 for all examples files to make it shorter and result web pages will be easier to review on small screen.

File: src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java [hunk 5]
```diff
@@ -34,36 +33,29 @@ protected String getPackageLocation() {
...
+        verifyWithInlineConfigParser(getPath("Example1.java"), expected);
     }
 
     @Test
     public void testExample2() throws Exception {
         final String[] expected = {
-
-        };
-
-        verifyWithInlineConfigParser(getPath("Example2.txt"), expected);
```
Comment (romani) [inline src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java:49]: why just removed ?

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java (hunk 5 above)
Comment (Coding-Aliens) [reply src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java:49]: I removed it, but the git treated it like an edit

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java (hunk 3 above)
Comment (Coding-Aliens) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java:21]: Done

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java (hunk 2 above)
Comment (Coding-Aliens) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java:22]: Done

File: src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java (hunk 5 above)
Comment (romani) [reply src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java:49]: Please restore it, or explain why you think it should not exists

Comment (Coding-Aliens) [reply src/xdocs-examples/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTokenTextCheckExamplesTest.java:49]: Oh, that's a mistake. I will restore it

File: src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java (hunk 4 above)
Comment (Coding-Aliens) [reply src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.java]: Done

**Comment by Coding-Aliens** [comment]
Ok, everything is done, thanks for your instruction!

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java [hunk 6]
```diff
@@ -9,9 +9,13 @@
...
 
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltokentext;
+
 // xdoc section -- start
-public void myTest() {
+public class Example1 {
+  public void myTest() {
     String test = "a href"; // violation
-    String test2 = "A href"; // OK, case is sensitive
+    // String test2 = "A href"; // OK, case is sensitive
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java:18]: uncomment this.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java (hunk 6 above)
Comment (Coding-Aliens) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java]: Finished

**Comment by Coding-Aliens** [comment]
OK, I get that rule

File: src/xdocs/checks/coding/illegaltokentext.xml [hunk 7]
```diff
@@ -137,9 +141,13 @@ public void myTest() {
         </source>
         <p id="Example3-code">Example:</p>
         <source>
-public void myTest() {
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltokentext;
+
+public class Example3 {
```
Comment (romani) [inline src/xdocs/checks/coding/illegaltokentext.xml:146]: Please put comment `// xdoc section -- start` in example avoid package being in user visible snippet

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/Example1.txt [hunk 8]
```diff
@@ -15,7 +15,7 @@ public class SomeClass {
   public SomeClass(String testField) { // violation, 'testField' param hides 'testField' field
   }
   public void method(String param) { // OK
-      String field = param; // violation, 'field' variable hides 'field' field
+    String field = param; // violation, 'field' variable hides 'field' field
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/Example1.txt:18]: Revert this change, we supposed to do update only for one Check

File: src/xdocs/checks/coding/illegaltokentext.xml.template [hunk 9]
```diff
@@ -61,13 +61,13 @@
         </p>
         <macro name="example">
           <param name="path"
-                 value="resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example3.txt"/>
```
Comment (romani) [inline src/xdocs/checks/coding/illegaltokentext.xml.template:64]: Please change description from
>To configure the check to forbid string literal text blocks containing
          <code>"""</code>:

To
```
To configure the check to forbid string literal text blocks containing
          <code>"</code>:
```

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java [hunk 10]
```diff
@@ -9,9 +9,13 @@
...
 
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltokentext;
+
 // xdoc section -- start
-public void myTest() {
+public class Example1 {
+  public void myTest() {
     String test = "a href"; // violation
     String test2 = "A href"; // OK, case is sensitive
+  }
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java:19]: Please make it ( extra space before =)
`String test  = "a href"; // violation`

To let comments to be vertically aligned

**Review by romani — CHANGES_REQUESTED** [review]
Last items

**Comment by Coding-Aliens** [comment]
All Updated

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
```
String test = "a href";  // violation
String test2 = "A href"; // OK, case is sensitive
```

Not aligned, please try to change name of variable if space is not working

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java (hunk 10 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltokentext/Example1.java:19]: Not done

**Comment by Coding-Aliens** [comment]
Done

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
please do same vertical alignment for Example2

**Comment by Coding-Aliens** [comment]
Done, I think for example4 its not needed as there is no comparison

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@Coding-Aliens , thanks lot, please do more contributions

File: src/xdocs/checks/coding/illegaltokentext.xml (hunk 7 above)
Comment (romani) [reply src/xdocs/checks/coding/illegaltokentext.xml:146]: @Coding-Aliens , please fix this in separate PR.
Comment should be right above a class, to make sure package is not showing in xdoc file.

**Comment by Coding-Aliens** [comment]
Alright.

**Comment by romani** [comment]
@Coding-Aliens, please address https://github.com/checkstyle/checkstyle/pull/14235#discussion_r1449099098

**Comment by Coding-Aliens** [comment]
Received now

File: src/xdocs/checks/coding/illegaltokentext.xml (hunk 7 above)
Comment (Coding-Aliens) [reply src/xdocs/checks/coding/illegaltokentext.xml:146]: Received
